src/model/project/procedure.rs

```diff
@@ -9,71 +9,97 @@ pub struct Procedure {
     pub condition: String,
     pub deploy_path: String,
     pub branches: Vec<String>,
+    pub log: Option<String>,
 }
 
 impl Procedure {
-    pub fn new(raw_procedure: &Value, raw_default_deploy_path: &Value) -> Result<Procedure, Error> {
-        let raw_name: &Value = &raw_procedure["name"];
-        if !raw_name.is_string() {
-            return Err(err_msg("Name is invalid in procedure"));
-        }
-        let name: &str = raw_name.as_str().unwrap();
+    pub fn new(raw_procedure: &Value, raw_default_deploy_path: Option<&Value>) -> Result<Procedure, Error> {
+        let name: &str = match raw_procedure.get("name") {
+            Some(raw_name) => match raw_name.as_str() {
+                Some(s) => s,
+                None => return Err(err_msg("Name is invalid in procedure")),
+            },
+            None => return Err(err_msg("Name not found in procedure")),
+        };
```

Comment:
This code seems way more verbose than it needs to be.

Reply:
If you want more specific errors then I guess it is fine.

Reply:
It expands everything so it's easier to understand what's actually happening with the values. Idk, I find this easier to read than as_str().unwrap().to_string() things

Reply:
In terms of performance, LLVM should handle these match statements the same (I hope). I get for readability you want all handling to be in one "block" for manageability. Guess it is time for me to use match statements a little more.